engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/BeanMetaDataImpl.java

```diff
@@ -465,26 +499,30 @@ public String toString() {
 		private DefaultGroupSequenceProvider<? super T> defaultGroupSequenceProvider;
 
 		private final MethodValidationConfiguration methodValidationConfiguration;
-		
+
 		private BeanMetaDataBuilder(
-				ConstraintHelper constraintHelper, 
-				ExecutableHelper executableHelper, 
+				ConstraintHelper constraintHelper,
+				ExecutableHelper executableHelper,
+				ValidationOrderGenerator validationOrderGenerator,
```

Comment:
First I was wondering why you were passing `ValidationOrderGenerator` around, since I thought it was stateless, but it is actually not. It caches the resolved sequences. Wondering about the use of `ConcurrentMap` in `ValidationOrderGenerator` . Maybe it should be a `ConcurrentReferenceHashMap` as well?